src/main/webapp/app/exercises/text/manage/example-text-submission/example-text-submission.component.html

```diff
@@ -97,7 +97,7 @@ <h5 *ngIf="toComplete" class="fw-light mt-3">{{ 'artemisApp.exampleSubmission.ho
         *ngIf="state.ui === UIStates.ASSESSMENT"
         [submission]="submission!"
         [(textBlockRefs)]="textBlockRefs"
-        [readOnly]="readOnly || (!exercise?.isAtLeastEditor && !toComplete)"
+        [readOnly]="readOnly || (!!exercise && !exercise.isAtLeastEditor && !toComplete)"
```

Comment:
```suggestion
        [readOnly]="readOnly || (exercise && !exercise.isAtLeastEditor && !toComplete)"
```
Wouldn't that be enough?

Reply:
This can be a problem when exercise has the type Exercise | undefined.
The double !! makes sure that the types are correct, since this will always produce a boolean

Reply:
That's right. Maybe I understand wrongly how that resolves. My understanding is:

exercise of type Exercise => is an object => `exercise` (true-ish); `!exercise` (false); `!!exercise` (true)
exercise undefined => `exercise` (false-ish); `!exercise` (true); `!!exercise` (false)

In my opinion, this should be enough. Where do you see the problem? Or does the linter complain? That's another story.

Reply:
Exactly, the linter does not like it without the !!